File: app/src/main/java/com/relic/data/gateway/PostGateway.java
package com.relic.data.gateway;

import android.arch.lifecycle.LiveData;

public interface PostGateway {
  LiveData<Boolean> voteOnPost(String fullname, int voteStatus);

  LiveData<Boolean> savePost(String fullname, boolean saved);

  LiveData<Boolean> comment(String fullname, String comment);

  LiveData<Boolean> gildPost(String fullname, boolean gild);

  LiveData<Boolean> reportPosts(String fullname, boolean report);

  LiveData<Boolean> visitPost(String postFullname);
}

File: app/src/main/java/com/relic/data/gateway/PostGateway.kt
package com.relic.data.gateway

import com.relic.api.response.Listing
import com.relic.data.PostRepository
import com.relic.domain.models.ListingItem

interface PostGateway {
    suspend fun voteOnPost(fullname: String, voteStatus: Int)

    suspend fun savePost(fullname: String, save: Boolean)

    suspend fun comment(fullname: String, comment: String)

    suspend fun gildPost(fullname: String, gild: Boolean)

    suspend fun reportPosts(fullname: String, report: Boolean)

    suspend fun visitPost(postFullname: String)
}
